src/ripple/app/tx/impl/CashCheck.cpp

```diff
@@ -195,16 +196,31 @@ CashCheck::preclaim(PreclaimContext const& ctx)
         // An issuer can always accept their own currency.
         if (!value.native() && (value.getIssuer() != dstId))
         {
+            auto const sleIssuer = ctx.view.read(keylet::account(issuerId));
             auto const sleTrustLine =
                 ctx.view.read(keylet::line(dstId, issuerId, currency));
+
             if (!sleTrustLine)
             {
-                JLOG(ctx.j.warn())
-                    << "Cannot cash check for IOU without trustline.";
-                return tecNO_LINE;
+                if (!ctx.view.rules().enabled(featureCheckCashMakesTrustLine))
```

Comment:
One way to do the changes from line 200 (in the original code) to line 225 that requires fewer actual code changes is to just add `&& !ctx.view.rules().enabled(featureCheckCashMakesTrustLine)` to the condition on line 200, and insert this condition on line 218:

```cpp
if (!sleTrustLine) {
    return tecNO_AUTH;
}
```

Reply:
Here it is as a commit: https://github.com/thejohnfreeman/rippled/commit/d57c0e801dff73946ee69faeb641d66117c32312